pkgs/top-level/all-packages.nix

```
@@ -17908,6 +17908,22 @@ in
     ];
   };
 
+  # Needs a `date` and a `sha256` argument. If version differs from `linux_testing` it also needs
+  # `extraMeta.branch` and `extraMeta.rc`.
+  linux_next = args: with lib; callPackage ../os-specific/linux/kernel/linux-next.nix (rec {
+    extraMeta.branch = args.extraMeta.branch or linux_testing.meta.branch;
+    extraMeta.rc = args.extraMeta.rc or (last (splitString "-" linux_testing.version));
+    kernelPatches = args.kernelPatches or linux_testing.kernelPatches;
+    sha256 = args.sha256 or lib.fakeSha256;
+    version = args.version or (
```

```suggestion
    extraMeta.branch = linux_testing.meta.branch;
    extraMeta.rc = last (splitString "-" linux_testing.version);
    kernelPatches = linux_testing.kernelPatches;
    sha256 = lib.fakeSha256;
    version = (
```
Since these will get overridden by `// args` anyway, we don't need the `or`s.

---

I think, the arguments of `//` need to be flipped for that to work.

I remember why I did it this way: this was the way I found to have `manually set values` > `default values` and still include all other possible fields from args. 

Going to test this (flipped version) while still trying to provide a default for all values.

---

After a bit of playing around the picture is clearer:
```.nix
{linux_next = args: with lib;
  let
    defaultArgs = {
      extraMeta.branch = linux_testing.meta.branch;
      extraMeta.rc = last (splitString "-" linux_testing.version);
      kernelPatches = linux_testing.kernelPatches;
      sha256 = fakeSha256;
    };
    combined = recursiveUpdate defaultArgs args;
    realArgs = { inherit version; } // combined;
    version =
      let branch = combined.extraMeta.branch + ".0";
          rc = combined.extraMeta.rc;
      in concatStringsSep "-" (
        # X.Y.0-rcZ-next
        [branch] ++ optional (rc != "") rc ++ ["next"]
      );
  in callPackage ../os-specific/linux/kernel/linux-next.nix realArgs;}
  ```
1. We need a `recursiveUpdate`to get all the _primary_ arguments right right,
2. so we can calculate the `version` field for the final attributes.

If this is an easier to understand approach, I'll replace the code in the PR.